CoreScriptsRoot/ServerCoreScripts/ClientChat/TransparencyTweener.lua

```diff
@@ -0,0 +1,196 @@
+local source = [[
+local module = {}
+
+local RunService = game:GetService("RunService")
+--////////////////////////////// Include
+--//////////////////////////////////////
+local modulesFolder = script.Parent
+
+--////////////////////////////// Details
+--//////////////////////////////////////
+local metatable = {}
+metatable.__ClassName = "TransparencyTweener"
+
+metatable.__tostring = function(tbl)
+	return tbl.__ClassName .. ": " .. tbl.MemoryLocation
+end
+
+metatable.__metatable = "The metatable is locked"
+metatable.__index = function(tbl, index, value)
+	if rawget(tbl, index) then return rawget(tbl, index) end
+	if rawget(metatable, index) then return rawget(metatable, index) end
+
+	if (index == "Transparency") then
+		return rawget(tbl, "InternalLastTweenPercentage")
+	end
+
+	error(index .. " is not a valid member of " .. tbl.__ClassName)
+end
+metatable.__newindex = function(tbl, index, value)
+	if (index == "Transparency") then
+		tbl.InternalLastTweenPercentage = value
+		tbl:SetPropertiesToTweenPercentage(value)
+		return
+	end
+
+	error(index .. " is not a valid member of " .. tbl.__ClassName)
+end
+
+
+--////////////////////////////// Methods
+--//////////////////////////////////////
+function metatable:Dump()
+	local str = tostring(self)
+
+	for tweenObject, objectProperties in pairs(self.TweenObjects) do
+		local addStr = " | "
+		if (type(tweenObject) == "table") then
+			addStr = addStr .. "{{" .. tweenObject:Dump() .. "}}"
+		elseif (type(tweenObject) == "userdata") then
+			addStr = addStr .. tweenObject.Name .. "/" .. (tweenObject.Parent and tweenObject.Parent.Name or nil)
+		end
+
+		for propertyName, baseValue in pairs(objectProperties) do
+			addStr =  addStr .. " [" .. propertyName .. "=" .. string.sub(tostring(baseValue), 1, 4) .. "]"
+		end
+
+		str = str .. addStr
+	end
+
+	return str
+end
+
+function metatable:OutputTest()
```

Comment:
I actually really like that you left this in. It'll be nice for whoever needs to maintain this in the future if you're not around to help. Maybe we should do this sort of thing more.